refl1d/reflectivity.py

```diff
@@ -285,33 +298,340 @@ def calculate_u1_u3_py(H, rhoM, thetaM, Aguide):
+        DBB = (BL - BLP) * DELTA; # multiply by delta here?
+        DBG = (1.0 - BL*GLP) * DELTA
+        DGB = (1.0 - GL*BLP) * DELTA
+        DGG = (GL - GLP) * DELTA
+
+        ES1L = exp(S1L*Z)
+        ENS1L = CR / ES1L
+        ES1LP = exp(S1LP*Z)
+        ENS1LP = CR / ES1LP
+        ES3L = exp(S3L*Z)
+        ENS3L = CR / ES3L
+        ES3LP = exp(S3LP*Z)
+        ENS3LP = CR / ES3LP
+
+        FS1S1 = S1L/S1LP
+        FS1S3 = S1L/S3LP
+        FS3S1 = S3L/S1LP
+        FS3S3 = S3L/S3LP
+
+        A11 = A22 = DBG * (1.0 + FS1S1)
+        A11 *= ES1L * ENS1LP
+        A22 *= ENS1L * ES1LP
+        A12 = A21 = DBG * (1.0 - FS1S1) * exp(2.*S1L*S1LP*SIGMAL*SIGMAL)
+        A12 *= ENS1L * ENS1LP
+        A21 *= ES1L  * ES1LP
+        A13 = A24 = DGG * (1.0 + FS3S1)
+        A13 *= ES3L  * ENS1LP
+        A24 *= ENS3L * ES1LP
+        A14 = A23 = DGG * (1.0 - FS3S1) * exp(2.*S3L*S1LP*SIGMAL*SIGMAL)
+        A14 *= ENS3L * ENS1LP
+        A23 *= ES3L  * ES1LP
+
+        A31 = A42 = DBB * (1.0 + FS1S3)
+        A31 *= ES1L * ENS3LP
+        A42 *= ENS1L * ES3LP
+        A32 = A41 = DBB * (1.0 - FS1S3) * exp(2.*S1L*S3LP*SIGMAL*SIGMAL)
+        A32 *= ENS1L * ENS3LP
+        A41 *= ES1L  * ES3LP
+        A33 = A44 = DGB * (1.0 + FS3S3)
+        A33 *= ES3L * ENS3LP
+        A44 *= ENS3L * ES3LP
+        A34 = A43 = DGB * (1.0 - FS3S3) * exp(2.*S3L*S3LP*SIGMAL*SIGMAL)
+        A34 *= ENS3L * ENS3LP
+        A43 *= ES3L * ES3LP
+
+
+        #    Matrix update B=A*B
+        C1=A11*B11+A12*B21+A13*B31+A14*B41
+        C2=A21*B11+A22*B21+A23*B31+A24*B41
+        C3=A31*B11+A32*B21+A33*B31+A34*B41
+        C4=A41*B11+A42*B21+A43*B31+A44*B41
+        B11=C1
+        B21=C2
+        B31=C3
+        B41=C4
+
+        C1=A11*B12+A12*B22+A13*B32+A14*B42
+        C2=A21*B12+A22*B22+A23*B32+A24*B42
+        C3=A31*B12+A32*B22+A33*B32+A34*B42
+        C4=A41*B12+A42*B22+A43*B32+A44*B42
+        B12=C1
+        B22=C2
+        B32=C3
+        B42=C4
+
+        C1=A11*B13+A12*B23+A13*B33+A14*B43
+        C2=A21*B13+A22*B23+A23*B33+A24*B43
+        C3=A31*B13+A32*B23+A33*B33+A34*B43
+        C4=A41*B13+A42*B23+A43*B33+A44*B43
+        B13=C1
+        B23=C2
+        B33=C3
+        B43=C4
+
+        C1=A11*B14+A12*B24+A13*B34+A14*B44
+        C2=A21*B14+A22*B24+A23*B34+A24*B44
+        C3=A31*B14+A32*B24+A33*B34+A34*B44
+        C4=A41*B14+A42*B24+A43*B34+A44*B44
+        B14=C1
+        B24=C2
+        B34=C3
+        B44=C4
+
+        Z += D[LP]
+        L = LP
+    
+    #    Done computing B = A(N)*...*A(2)*A(1)*I
+    DETW=(B44*B22-B24*B42)
+
+
+    #    Calculate reflectivity coefficients specified by POLSTAT
+    YA[i] = (B24*B41-B21*B44)/DETW # ++
+    YB[i] = (B21*B42-B41*B22)/DETW # +-
+    YC[i] = (B24*B43-B23*B44)/DETW # -+
+    YD[i] = (B23*B42-B43*B22)/DETW # --
+
+@cc.export('mag_amplitude', 'void(f8[:], f8[:], f8[:], f8[:], f8[:], c16[:], c16[:], f8, f8[:], i4[:], c16[:], c16[:], c16[:], c16[:])')
+@njit(parallel=True, cache=True)
```

Comment:
Put the type signature with the `njit` rather than the `cc`.

Reply:
I wasn't aware that this would work!